include/autocheck/generator_combinators.hpp

```diff
@@ -45,10 +46,11 @@ namespace autocheck {
           pred(pred), gen(gen) {}
 
         result_type operator() (size_t size = 0) {
-          while (true) {
+          for(size_t i{0}; i < std::numeric_limits<size_t>::max(); ++i){
```

Comment:
I merged this from another fork and didn't fully understand the generator at the time.